src/connection/tcp.rs

```diff
@@ -39,7 +42,15 @@ pub fn open_secret_connection(
     socket.set_read_timeout(Some(timeout))?;
     socket.set_write_timeout(Some(timeout))?;
 
-    let connection = SecretConnection::new(socket, &identity_key, protocol_version)?;
+    let connection = match SecretConnection::new(socket, &identity_key, protocol_version) {
+        Ok(conn) => conn,
+        Err(error) => match error.downcast_ref::<TmError>() {
```

Comment:
this is ugly... should I attempt to rewrite this part to use `eyre`?

Reply:
That's fine for now.

This is a tradeoff with using `eyre` to aggregate error types. If you want to handle them, you either need to use downcasting like this to get the concrete type, or you need to return a `Result` with the underlying error enum.